qa/tasks/quiescer.py

```diff
@@ -0,0 +1,295 @@
+"""
+Thrash mds by randomly quiescing the fs root
+"""
+import logging
+import contextlib
+
+from teuthology import misc as teuthology
+
+from tasks import ceph_manager
+from tasks.cephfs.filesystem import MDSCluster, Filesystem
+from tasks.thrasher import ThrasherGreenlet
+
+import random
+import math
+import errno
+import json
+import time
+
+from io import StringIO
+
+log = logging.getLogger(__name__)
+
+class Quiescer(ThrasherGreenlet):
+    """
+    The Quiescer does periodic quiescing of the configured paths, by default - the root '/'.
+
+    quiesce_timeout: [1..),default=30 :: maximum time in seconds to wait for the quiesce to succeed
+    quiesce_factor: [0.005..0.5],default=0.1 :: the fraction of the total runtime we want the system quiesced
+    min_quiesce: [1..),default=10 :: the minimum pause time in seconds
+    max_quiesce: [1..),default=60 :: the maximum pause time in seconds
+    initial_delay: [0..),default=120 :: the time in seconds before the first quiesce
+    seed: default=None :: an optional seed to a pseudorandom sequence of quiesce durations
+    roots: List[String], default=["/"] :: the roots to quiesce
+    cancelations_cap: [-1..),default=10 :: the number of times we ignore canceled quiesce sets
+    split_if_longer: int,default=mean(min_quiesce, max_quiesce) :: if the duration is longer than this,
```

Comment:
duration is the expiration time of the quiesce (how long the root should be quiesced)?

Reply:
Yes. This would be a single long quiesce if this property is set to a higher value, otherwise, it's split into two half-time quiesces. The rationale is to cover the case when a quiesce is started shortly after the previous one has expired.